src/roiextractors/extractors/femtonicsimagingextractor/femtonicsimagingextractor.py

```diff
@@ -0,0 +1,418 @@
+
+    def _setup_video_data(self):
+        """Set up access to the actual imaging data for the selected session, unit, and channel."""
+        # Find the channel index by looking for the channel name in the HDF5 structure
+        available_channels = self._get_available_channels_from_file()
+        channel_index = available_channels.index(self._selected_channel_name)
+        channel_key = f"Channel_{channel_index}"
+
+        channel_path = f"{self._session_key}/{self._munit_key}/{channel_key}"
+
+        if channel_key in self._file_handle[self._session_key][self._munit_key]:
+            self._video = DatasetView(self._file_handle[self._session_key][self._munit_key][channel_key])
+        else:
+            raise Exception(f"Cannot find data at {channel_path}")
+
+    def _setup_sampling_frequency(self):
+        """Cache the sampling frequency during initialization to avoid repeated file access."""
+        time_per_frame_ms = self._get_frame_duration_in_milliseconds()
+        if time_per_frame_ms is None:
+            raise ValueError(
+                f"Sampling frequency could not be determined from metadata in "
+                f"{self._session_key}/{self._munit_key}."
+            )
+        self._sampling_frequency = 1000.0 / time_per_frame_ms
+
+    def _get_available_channels_from_file(self) -> list[str]:
+        """Get available channels from the current session/unit combination."""
+        if self._session_key not in self._file_handle or self._munit_key not in self._file_handle[self._session_key]:
+            raise ValueError(f"Session {self._session_index}/Unit {self._munit_index} not found in file")
+
+        attrs = dict(self._file_handle[self._session_key][self._munit_key].attrs)
+        num_channels = attrs.get("VecChannelsSize", 2)
+
+        channels = []
+        for i in range(num_channels):
+            name_arr = attrs.get(f"Channel_{i}_Name", [])
+            name = self._decode_string(name_arr) if len(name_arr) > 0 else f"Channel_{i}"
+            channels.append(name)
+        return channels
+
+    def _decode_string(self, arr) -> str:
+        """Convert int16 array to string."""
+        if len(arr) == 0:
+            return ""
+        return "".join(chr(x) for x in arr if x != 0)
+
+    def get_image_shape(self) -> tuple[int, int]:
+        """Get the shape of the video frame (num_rows, num_columns)."""
+        return self._video.shape[1], self._video.shape[2]  # (height, width)
+
+    def get_num_samples(self) -> int:
+        """Get the number of samples (frames) in the video."""
+        return self._video.shape[0]
+
+    def get_channel_names(self) -> list[str]:
+        """Get the channel names."""
+        return [self._selected_channel_name]
+
+    def get_sampling_frequency(self) -> float:
+        """Get the sampling frequency in Hz."""
+        return self._sampling_frequency
+
+    def _get_frame_duration_in_milliseconds(self) -> Optional[float]:
```

Comment:
Here on the other hand, we do need th check of the `ZAxisConversionUnitName` and throw an error if it is not in milliseconds. It is very important that get the sampling frequency in Hz correclty.